src/main/java/org/jitsi/videobridge/Endpoint.java

```diff
@@ -525,59 +537,92 @@ private void onPinnedEndpointChangedEvent(
             sc.bind("this", this);
             logger.debug(sc.c(
                     "Endpoint {this.id} notified us that it has pinned"
-                        + " {pinnedId}."));
+                            + " {pinnedId}."));
         }
 
-        Conference conference = weakConference.get();
+        pinnedEndpointsChanged(Collections.singletonList(newPinnedEndpointID));
```

Comment:
Here's how I fixed the bug mentioned in my previous comment: 

List<String> newPinnedIDList = (newPinnedEndpointID == null) ? Collections.EMPTY_LIST : Collections.singletonList(newPinnedEndpointID);
        pinnedEndpointsChanged(newPinnedIDList);